src/test/unit/test_lumigo_utils.py

```diff
@@ -625,3 +626,18 @@ def test_internal_analytics_message(capsys):
     assert capsys.readouterr().out.startswith(INTERNAL_ANALYTICS_PREFIX)
     internal_analytics_message("Message")
     assert capsys.readouterr().out == ""
+
+
+@pytest.mark.parametrize(
+    "old, new, expected",
+    [
+        (b"1", b"2", b"12"),  # happy flow
+        (b"", b"2", b"2"),  # no old
+        (b"1", b"", b"1"),  # no new
+        (b"12", b"34567", b"12345...[too long]"),  # together pass max size
+        (b"123456", b"789", b"12345...[too long]"),  # old pass max size
```

Comment:
Add a test with json and in the json there's a string with the `"` inside the string.
Something like: `json.dumps({"k": 'h"ell"o'})`